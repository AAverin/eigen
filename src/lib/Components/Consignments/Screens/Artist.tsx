import * as React from "react"

import ArtistSearch from "../Components/ArtistSearchResults"
import DoneButton from "../Components/BottomAlignedButton"
import ConsignmentBG from "../Components/ConsignmentBG"

import { ArtistResult, ConsignmentSetup } from "../index"

import { debounce } from "lodash"
import { NavigatorIOS, Route, ScrollView, View, ViewProperties } from "react-native"
import metaphysics from "../../../metaphysics"

interface ArtistSearchResponse {
  match_artist: ArtistResult[]
}

interface Props extends ConsignmentSetup {
  navigator: NavigatorIOS
  route: Route
  updateWithResult?: (result: ArtistResult) => void
}

interface State {
  query: string
  searching: boolean
  results: any | null
}

export default class Artist extends React.Component<Props, State> {
  constructor(props) {
    super(props)
    this.state = {
      query: null,
      searching: false,
      results: null,
    }
  }

  doneTapped = () => {
    this.props.navigator.pop()
  }

  artistSelected = (result: ArtistResult) => {
    this.props.navigator.pop()
    this.props.updateWithResult(result)
  }

  textChanged = async (text: string) => {
    this.setState({ query: text, searching: text.length > 0 })
    this.searchForQuery(text)
  }

  searchForQuery = async (query: string) => {
    const results = await metaphysics<ArtistSearchResponse>(`
      {
        match_artist(term: "${query}") {
          id
          name
          image {
            url
          }
        }
      }
    `)
    this.setState({ results: results.match_artist, searching: false })
  }

  render() {
    // This might become a higher order component for reuse, if used more elsewhere
    const doneButtonStyles = {
      backgroundColor: "black",
      marginBottom: 20,
      paddingTop: 18,
      height: 56,
    }

    return (
      <ConsignmentBG>
        <DoneButton onPress={this.doneTapped} bodyStyle={doneButtonStyles} buttonText="DONE">
          <View
            style={{ alignContent: "center", justifyContent: "flex-end", flexGrow: 1, marginLeft: 20, marginRight: 20 }}
          >
            <ArtistSearch
              results={this.state.results}
              query={this.state.query}
              onChangeText={this.textChanged}
              searching={this.state.searching}
              resultSelected={this.artistSelected}
            />
          </View>
        </DoneButton>
      </ConsignmentBG>
    )
  }
}
