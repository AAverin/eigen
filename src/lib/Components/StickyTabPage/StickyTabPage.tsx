import { color, Spacer } from "@artsy/palette"
import { Schema } from "lib/utils/track"
import { useScreenDimensions } from "lib/utils/useScreenDimensions"
import React, { useMemo, useState } from "react"
import { View } from "react-native"
import Animated from "react-native-reanimated"
import { useTracking } from "react-tracking"
import { useAnimatedValue } from "./reanimatedHelpers"
import { SnappyHorizontalRail } from "./SnappyHorizontalRail"
import { StickyTabPageTabBar } from "./StickyTabPageTabBar"
import { StickyTabScrollView } from "./StickyTabScrollView"

interface TabProps {
  initial?: boolean
  title: string
  content: JSX.Element
}

/**
 * This page wrapper encapsulates a disappearing header and sticky tabs each with their own content
 *
 * At the moment all tabs are rendered at all times, as this isn't designed for more than 3 tabs
 * but if we need to have conditional rendering of tab content in the future it should be possible.
 *
 * Each tab optionally consumes a 'scroll view context' which could potentialy contain information
 * about whether the tab is being shown currently etc.
 */
export const StickyTabPage: React.FC<{
  tabs: TabProps[]
  headerContent: JSX.Element
}> = ({ tabs, headerContent }) => {
  const { width } = useScreenDimensions()
  const initialActiveIndex = useMemo(() => Math.max(tabs.findIndex(tab => tab.initial), 0), [])
  const [activeTabIndex, setActiveTabIndex] = useState(initialActiveIndex)
  const [headerHeight, setHeaderHeight] = useState<null | number>(null)
  const tracking = useTracking()
  const headerOffsetY = useAnimatedValue(0)

  return (
    <View style={{ flex: 1, position: "relative", overflow: "hidden" }}>
      {/* put tab content first because we want the header to be absolutely positioned _above_ it */}
      {headerHeight !== null && (
        <SnappyHorizontalRail offset={activeTabIndex * width}>
          {tabs.map(({ content }, index) => {
            return (
              <View style={{ flex: 1, width }} key={index}>
                <StickyTabScrollView
                  headerHeight={headerHeight}
                  content={content}
                  headerOffsetY={headerOffsetY}
                  isActive={index === activeTabIndex}
                />
              </View>
            )
          })}
        </SnappyHorizontalRail>
      )}
      <Animated.View
        style={{
          width,
          top: 0,
          position: "absolute",
          backgroundColor: color("white100"),
          transform: [{ translateY: headerOffsetY as any }],
        }}
      >
        <View onLayout={e => setHeaderHeight(e.nativeEvent.layout.height)}>
          {headerContent}
          <Spacer mb={1} />
        </View>
        <StickyTabPageTabBar
          labels={tabs.map(({ title }) => title)}
          initialActiveIndex={initialActiveIndex}
          onIndexChange={index => {
            setActiveTabIndex(index)
            tracking.trackEvent({
              action_name: tabs[index].title,
              action_type: Schema.ActionTypes.Tap,
            })
          }}
        />
      </Animated.View>
    </View>
  )
}
