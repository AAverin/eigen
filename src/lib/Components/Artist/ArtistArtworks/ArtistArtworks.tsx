import { Box, Separator, Spacer } from "@artsy/palette"
import { ArtistArtworks_artist } from "__generated__/ArtistArtworks_artist.graphql"
import { ArtistNotableWorksRailFragmentContainer } from "lib/Components/Artist/ArtistArtworks/ArtistNotableWorksRail"
import {
  InfiniteScrollArtworksGridContainer as InfiniteScrollArtworksGrid,
  Props as InfiniteScrollGridProps,
} from "lib/Components/ArtworkGrids/InfiniteScrollArtworksGrid"
import { StickyTabPageScrollView } from "lib/Components/StickyTabPage/StickyTabPageScrollView"
import React from "react"
import { createPaginationContainer, graphql, RelayPaginationProp } from "react-relay"
import { ArtistCollectionsRailFragmentContainer } from "./ArtistCollectionsRail"

interface Props extends InfiniteScrollGridProps {
  artist: ArtistArtworks_artist
  relay: RelayPaginationProp
}

const ArtworksGrid: React.FC<Props> = ({ artist, relay, ...props }) => (
  <StickyTabPageScrollView>
    <Spacer mb={2} />
    <ArtistNotableWorksRailFragmentContainer artist={artist} {...props} />
    <ArtistCollectionsRailFragmentContainer collections={artist.iconicCollections} artist={artist} {...props} />
    <Box mx={"-20px"} mb={3} mt={1}>
      <Separator />
    </Box>
    <InfiniteScrollArtworksGrid
      // @ts-ignore STRICTNESS_MIGRATION
      connection={artist.artworks}
      loadMore={relay.loadMore}
      hasMore={relay.hasMore}
      isLoading={relay.isLoading}
      {...props}
    />
  </StickyTabPageScrollView>
)

export default createPaginationContainer(
  ArtworksGrid,
  {
    artist: graphql`
      fragment ArtistArtworks_artist on Artist
        @argumentDefinitions(count: { type: "Int", defaultValue: 10 }, cursor: { type: "String" }) {
        id
        internalID
        artworks: filterArtworksConnection(
          first: $count
          after: $cursor
          sort: "-decayed_merch"
          aggregations: [TOTAL]
        ) @connection(key: "ArtistArtworksGrid_artworks") {
          # TODO: Just here to satisfy the relay compiler, can we get rid of this need?
          edges {
            node {
              id
            }
          }
          ...InfiniteScrollArtworksGrid_connection
        }

        ...ArtistCollectionsRail_artist

        iconicCollections: marketingCollections(isFeaturedArtistContent: true, size: 16) {
          ...ArtistCollectionsRail_collections
        }

        ...ArtistNotableWorksRail_artist
      }
    `,
  },
  {
    direction: "forward",
    getConnectionFromProps(props) {
      return props.artist && props.artist.artworks
    },
    getFragmentVariables(prevVars, totalCount) {
      return {
        ...prevVars,
        count: totalCount,
      }
    },
    getVariables(props, { count, cursor }, { filter }) {
      return {
        id: props.artist.id,
        count,
        cursor,
        filter,
      }
    },
    query: graphql`
      query ArtistArtworksQuery($id: ID!, $count: Int!, $cursor: String) {
        node(id: $id) {
          ... on Artist {
            ...ArtistArtworks_artist @arguments(count: $count, cursor: $cursor)
          }
        }
      }
    `,
  }
)
