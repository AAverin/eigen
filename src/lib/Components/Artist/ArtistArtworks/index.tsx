import { color, FilterIcon, Flex, Sans } from "@artsy/palette"
import { ArtistArtworks_artist } from "__generated__/ArtistArtworks_artist.graphql"
import {
  InfiniteScrollArtworksGridContainer as InfiniteScrollArtworksGrid,
  Props as InfiniteScrollGridProps,
} from "lib/Components/ArtworkGrids/InfiniteScrollArtworksGrid"
import { FilterModalNavigator } from "lib/Components/FilterModal"
import { StickyTabPageScrollView } from "lib/Components/StickyTabPage/StickyTabPageScrollView"
import { ArtworkFilterContext, ArtworkFilterGlobalStateProvider } from "lib/utils/ArtworkFiltersStore"
import React, { useState } from "react"
import { TouchableWithoutFeedback } from "react-native"
import { createPaginationContainer, graphql, RelayPaginationProp } from "react-relay"
import styled from "styled-components/native"
import { ArtistCollectionsRailFragmentContainer } from "./ArtistCollectionsRail"

// TODO: These should probably be shared components, also used for collections
// Should we replace existing FilterArtworkButton in "lib/Components/FilterModal"?
export const FilterArtworkButtonContainer = styled(Flex)`
  position: absolute;
  bottom: 20;
  flex: 1;
  justify-content: center;
  width: 100%;
  flex-direction: row;
`

export const FilterArtworkButton = styled(Flex)<{ isFilterCountVisible: boolean }>`
  border-radius: 20;
  background-color: ${color("black100")};
  align-items: center;
  justify-content: center;
  flex-direction: row;
  box-shadow: 0px 3px 3px rgba(0, 0, 0, 0.12);
`

interface ViewableItems {
  viewableItems?: ViewToken[]
}

interface ViewToken {
  item?: any
  key?: string
  index?: number | null
  isViewable?: boolean
  section?: any
}

interface ArtworksGridProps extends InfiniteScrollGridProps {
  artist: ArtistArtworks_artist
  relay: RelayPaginationProp
}

const ArtworksGrid: React.FC<ArtworksGridProps> = ({ artist, relay, ...props }) => {
  const [isFilterButtonVisible, setFilterButtonVisible] = useState(false)
  const [isFilterArtworksModalVisible, setFilterArtworkModalVisible] = useState(false)

  const viewabilityConfigRef = React.useRef({
    viewAreaCoveragePercentThreshold: 0, // What percentage of the artworks component should be in the screen before toggling the filter button
  })

  // TODO: This is only being called once, why?
  const onViewableItemsChangedRef = React.useRef((viewableItems: ViewableItems) => {
    return setFilterButtonVisible(!isFilterButtonVisible)
  })

  const handleFilterArtworksModal = () => {
    setFilterArtworkModalVisible(!isFilterArtworksModalVisible)
  }

  const openFilterArtworksModal = () => {
    handleFilterArtworksModal()
  }

  const closeFilterArtworksModal = () => {
    handleFilterArtworksModal()
  }

  return (
    <ArtworkFilterGlobalStateProvider>
      <ArtworkFilterContext.Consumer>
        {value => {
          return (
            <>
              <StickyTabPageScrollView
                onViewableItemsChanged={onViewableItemsChangedRef.current}
                viewabilityConfig={viewabilityConfigRef.current}
              >
                <ArtistCollectionsRailFragmentContainer
                  collections={artist.iconicCollections}
                  artist={artist}
                  {...props}
                />
                <InfiniteScrollArtworksGrid
                  // @ts-ignore STRICTNESS_MIGRATION
                  connection={artist.artworks}
                  loadMore={relay.loadMore}
                  {...props}
                />
                <FilterModalNavigator
                  {...props}
                  id={artist.id}
                  slug={artist.slug}
                  isFilterArtworksModalVisible={isFilterArtworksModalVisible}
                  exitModal={handleFilterArtworksModal}
                  closeModal={closeFilterArtworksModal}
                />
              </StickyTabPageScrollView>
              {isFilterButtonVisible && (
                <FilterArtworkButtonContainer>
                  <TouchableWithoutFeedback onPress={openFilterArtworksModal}>
                    <FilterArtworkButton
                      px="2"
                      isFilterCountVisible={value.state.appliedFilters.length > 0 ? true : false}
                    >
                      <FilterIcon fill="white100" />
                      <Sans size="3t" pl="1" py="1" color="white100" weight="medium">
                        Filter
                      </Sans>
                      {value.state.appliedFilters.length > 0 && (
                        <>
                          <Sans size="3t" pl={0.5} py="1" color="white100" weight="medium">
                            {"\u2022"}
                          </Sans>
                          <Sans size="3t" pl={0.5} py="1" color="white100" weight="medium">
                            {value.state.appliedFilters.length}
                          </Sans>
                        </>
                      )}
                    </FilterArtworkButton>
                  </TouchableWithoutFeedback>
                </FilterArtworkButtonContainer>
              )}
            </>
          )
        }}
      </ArtworkFilterContext.Consumer>
    </ArtworkFilterGlobalStateProvider>
  )
}

export default createPaginationContainer(
  ArtworksGrid,
  {
    artist: graphql`
      fragment ArtistArtworks_artist on Artist
        @argumentDefinitions(count: { type: "Int", defaultValue: 10 }, cursor: { type: "String" }) {
        id
        slug
        internalID
        artworks: filterArtworksConnection(
          first: $count
          after: $cursor
          sort: "-decayed_merch"
          aggregations: [TOTAL]
        ) @connection(key: "ArtistArtworksGrid_artworks") {
          # TODO: Just here to satisfy the relay compiler, can we get rid of this need?
          edges {
            node {
              id
            }
          }
          ...InfiniteScrollArtworksGrid_connection
        }

        ...ArtistCollectionsRail_artist

        iconicCollections: marketingCollections(isFeaturedArtistContent: true, size: 16) {
          ...ArtistCollectionsRail_collections
        }
      }
    `,
  },
  {
    direction: "forward",
    getConnectionFromProps(props) {
      return props.artist && props.artist.artworks
    },
    getFragmentVariables(prevVars, totalCount) {
      return {
        ...prevVars,
        count: totalCount,
      }
    },
    getVariables(props, { count, cursor }, { filter }) {
      return {
        id: props.artist.id,
        count,
        cursor,
        filter,
      }
    },
    query: graphql`
      query ArtistArtworksQuery($id: ID!, $count: Int!, $cursor: String) {
        node(id: $id) {
          ... on Artist {
            ...ArtistArtworks_artist @arguments(count: $count, cursor: $cursor)
          }
        }
      }
    `,
  }
)
