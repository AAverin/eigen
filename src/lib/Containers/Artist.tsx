import { Flex, Separator, Spacer, Theme } from "@artsy/palette"
import { Artist_artist } from "__generated__/Artist_artist.graphql"
import { ArtistQuery, ArtistQueryVariables } from "__generated__/ArtistQuery.graphql"
import ArtistAbout from "lib/Components/Artist/ArtistAbout"
import ArtistArtworks from "lib/Components/Artist/ArtistArtworks"
import ArtistHeader from "lib/Components/Artist/ArtistHeader"
import ArtistShows from "lib/Components/Artist/ArtistShows"
import { StickyTabPage } from "lib/Components/StickyTabPage/StickyTabPage"
import { defaultEnvironment } from "lib/relay/createEnvironment"
import { PlaceholderBox, PlaceholderImage, PlaceholderText } from "lib/utils/placeholders"
import { renderWithPlaceholder } from "lib/utils/renderWithPlaceholder"
import { track } from "lib/utils/track"
import { ProvideScreenDimensions } from "lib/utils/useScreenDimensions"
import React from "react"
import { ViewProperties } from "react-native"
import { createFragmentContainer, graphql, QueryRenderer } from "react-relay"

interface Props extends ViewProperties {
  artist: Artist_artist
}

// screen views are tracked in eigen
@track()
export class Artist extends React.Component<Props> {
  render() {
    const { artist } = this.props
    const tabs = []
    const displayAboutSection = artist.has_metadata || artist.counts.articles > 0 || artist.counts.related_artists > 0

    if (displayAboutSection) {
      tabs.push({
        title: "About",
        content: <ArtistAbout artist={artist} />,
      })
    }

    if (artist.counts.artworks) {
      tabs.push({
        title: "Artworks",
        initial: true,
        content: <ArtistArtworks artist={artist} />,
      })
    }

    if (artist.counts.partner_shows) {
      tabs.push({
        title: "Shows",
        content: <ArtistShows artist={artist} />,
      })
    }

    return (
      <Theme>
        <ProvideScreenDimensions>
          <Flex style={{ flex: 1 }}>
            <StickyTabPage headerContent={<ArtistHeader artist={artist} />} tabs={tabs} />
          </Flex>
        </ProvideScreenDimensions>
      </Theme>
    )
  }
}

export const ArtistFragmentContainer = createFragmentContainer(Artist, {
  artist: graphql`
    fragment Artist_artist on Artist {
      internalID
      slug
      has_metadata: hasMetadata
      counts {
        artworks
        partner_shows: partnerShows
        related_artists: relatedArtists
        articles
      }
      ...ArtistHeader_artist
      ...ArtistAbout_artist
      ...ArtistShows_artist
      ...ArtistArtworks_artist
    }
  `,
})

export const ArtistQueryRenderer: React.SFC<ArtistQueryVariables> = ({ artistID, isPad }) => {
  return (
    <QueryRenderer<ArtistQuery>
      environment={defaultEnvironment}
      query={graphql`
        query ArtistQuery($artistID: String!, $isPad: Boolean!) {
          artist(id: $artistID) {
            ...Artist_artist
          }
        }
      `}
      variables={{ artistID, isPad }}
      render={renderWithPlaceholder({
        Container: ArtistFragmentContainer,
        renderPlaceholder: () => <ArtistPlaceholder />,
      })}
    />
  )
}

const ArtistPlaceholder: React.FC = () => (
  <Theme>
    <Flex>
      <Flex alignItems="center">
        <Spacer mb={45} />
        {/* artist name */}
        <PlaceholderText width={120} />
        <Spacer mb={1} />
        {/* birthday, followers */}
        <PlaceholderText width={150} />
      </Flex>
      <Spacer mb={1} />
      {/* follow buton */}
      <PlaceholderBox height={42} marginHorizontal={20} />
      <Spacer mb={3} />
      {/* tabs */}
      <Flex justifyContent="space-around" flexDirection="row" px={2}>
        <PlaceholderText width={40} />
        <PlaceholderText width={50} />
        <PlaceholderText width={40} />
      </Flex>
      <Spacer mb={1} />
      <Separator />
      <Spacer mb={3} />
      {/* masonry grid */}
      <Flex mx={2} flexDirection="row">
        <Flex mr={1} style={{ flex: 1 }}>
          <PlaceholderImage height={92} />
          <PlaceholderImage height={172} />
          <PlaceholderImage height={82} />
        </Flex>
        <Flex ml={1} style={{ flex: 1 }}>
          <PlaceholderImage height={182} />
          <PlaceholderImage height={132} />
          <PlaceholderImage height={86} />
        </Flex>
      </Flex>
    </Flex>
  </Theme>
)
