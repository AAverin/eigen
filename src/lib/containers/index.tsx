import Artist from "./artist"
import Gene from "./gene"
import Home from "./home"

export default {
  Artist,
  Home,
  Gene,
}
