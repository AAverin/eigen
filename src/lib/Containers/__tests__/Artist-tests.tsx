import ArtistAbout from "lib/Components/Artist/ArtistAbout"
import ArtistArtworks from "lib/Components/Artist/ArtistArtworks"
import ArtistHeader from "lib/Components/Artist/ArtistHeader"
import ArtistShows from "lib/Components/Artist/ArtistShows"
import { StickyTab } from "lib/Components/StickyTabPage/StickyTabPageTabBar"
import _ from "lodash"
import React from "react"
import "react-native"
import { create } from "react-test-renderer"
import { createMockEnvironment, MockPayloadGenerator } from "relay-test-utils"
import { MockResolvers } from "relay-test-utils/lib/RelayMockPayloadGenerator"
import { ArtistQueryRenderer } from "../Artist"

jest.unmock("react-relay")

type ArtistQueries = "ArtistAboveTheFoldQuery" | "ArtistBelowTheFoldQuery"

describe("availableTabs", () => {
  let environment = createMockEnvironment()
  beforeEach(() => {
    environment = createMockEnvironment()
  })

  function mockMostRecentOperation(name: ArtistQueries, mockResolvers: MockResolvers = {}) {
    expect(environment.mock.getMostRecentOperation().request.node.operation.name).toBe(name)
    environment.mock.resolveMostRecentOperation(operation => {
      const result = MockPayloadGenerator.generate(operation, {
        ID({ path }) {
          if (_.isEqual(path, ["artist", "id"])) {
            return "artist-id"
          }
        },
        ...mockResolvers,
      })
      return result
    })
  }

  const TestWrapper = () => {
    return <ArtistQueryRenderer artistID="ignored" environment={environment} isPad={false} />
  }

  it("returns nothing if artist has no metadata, shows, or works", async () => {
    const tree = create(<TestWrapper />)
    mockMostRecentOperation("ArtistAboveTheFoldQuery", {
      Artist() {
        return {
          has_metadata: false,
          counts: { articles: 0, related_artists: 0, artworks: 0, partner_shows: 0 },
        }
      },
    })
    expect(tree.root.findAllByType(ArtistHeader)).toHaveLength(1)
    expect(tree.root.findAllByType(StickyTab)).toHaveLength(0)
  })

  it("returns About tab if artist has metadata", async () => {
    const tree = create(<TestWrapper />)
    mockMostRecentOperation("ArtistAboveTheFoldQuery", {
      Artist() {
        return {
          has_metadata: true,
          counts: { articles: 0, related_artists: 0, artworks: 0, partner_shows: 0 },
        }
      },
    })
    expect(tree.root.findAllByType(ArtistHeader)).toHaveLength(1)
    expect(tree.root.findAllByType(StickyTab)).toHaveLength(1)
    expect(tree.root.findAllByType(ArtistAbout)).toHaveLength(0)
    // it only shows below the fold
    mockMostRecentOperation("ArtistBelowTheFoldQuery")
    expect(tree.root.findAllByType(ArtistAbout)).toHaveLength(1)
  })

  it("returns About tab if artist has articles", async () => {
    const tree = create(<TestWrapper />)
    mockMostRecentOperation("ArtistAboveTheFoldQuery", {
      Artist() {
        return {
          has_metadata: false,
          counts: { articles: 1, related_artists: 0, artworks: 0, partner_shows: 0 },
        }
      },
    })
    mockMostRecentOperation("ArtistBelowTheFoldQuery")
    expect(tree.root.findAllByType(ArtistAbout)).toHaveLength(1)
  })

  it("returns Shows tab if artist has shows", async () => {
    const tree = create(<TestWrapper />)
    mockMostRecentOperation("ArtistAboveTheFoldQuery", {
      Artist() {
        return {
          has_metadata: false,
          counts: { articles: 0, related_artists: 0, artworks: 0, partner_shows: 1 },
        }
      },
    })
    mockMostRecentOperation("ArtistBelowTheFoldQuery")
    expect(tree.root.findAllByType(ArtistShows)).toHaveLength(1)
  })

  it("returns all three tabs if artist has metadata, works, and shows", async () => {
    const tree = create(<TestWrapper />)
    mockMostRecentOperation("ArtistAboveTheFoldQuery", {
      Artist() {
        return {
          has_metadata: true,
          counts: { articles: 1, related_artists: 0, artworks: 1, partner_shows: 1 },
        }
      },
    })
    expect(tree.root.findAllByType(ArtistArtworks)).toHaveLength(1)
    mockMostRecentOperation("ArtistBelowTheFoldQuery")
    expect(tree.root.findAllByType(ArtistAbout)).toHaveLength(1)
    expect(tree.root.findAllByType(ArtistShows)).toHaveLength(1)
  })
})
