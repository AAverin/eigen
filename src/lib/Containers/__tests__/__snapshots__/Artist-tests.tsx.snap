// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout works as expected with no tabs 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
      },
      Object {
        "flex": 1,
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFF",
          "position": "absolute",
          "top": 0,
          "transform": Array [
            Object {
              "translateY": Object {
                "setValue": [Function],
              },
            },
          ],
          "width": 380,
        }
      }
    >
      <View
        onLayout={[Function]}
      >
        <ArtistHeader
          artist={
            Object {
              "birthday": null,
              "counts": Object {
                "articles": 0,
                "artworks": 0,
                "partner_shows": 0,
              },
              "current_shows": null,
              "has_metadata": false,
              "id": null,
              "internalID": null,
              "name": null,
              "nationality": null,
              "past_large_shows": null,
              "past_small_shows": null,
              "upcoming_shows": null,
            }
          }
        />
        <View
          mb={1}
          style={
            Array [
              Object {
                "marginBottom": 10,
              },
            ]
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "#C2C2C2",
              "borderBottomWidth": 1,
              "display": "flex",
              "flexDirection": "row",
              "height": 48,
              "paddingLeft": 20,
              "paddingRight": 20,
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`layout works as expected with one tab 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
      },
      Object {
        "flex": 1,
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFF",
          "position": "absolute",
          "top": 0,
          "transform": Array [
            Object {
              "translateY": Object {
                "setValue": [Function],
              },
            },
          ],
          "width": 380,
        }
      }
    >
      <View
        onLayout={[Function]}
      >
        <ArtistHeader
          artist={
            Object {
              "birthday": null,
              "counts": Object {
                "articles": 0,
                "artworks": 0,
                "partner_shows": 0,
              },
              "current_shows": null,
              "has_metadata": true,
              "id": null,
              "internalID": null,
              "name": null,
              "nationality": null,
              "past_large_shows": null,
              "past_small_shows": null,
              "upcoming_shows": null,
            }
          }
        />
        <View
          mb={1}
          style={
            Array [
              Object {
                "marginBottom": 10,
              },
            ]
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "#C2C2C2",
              "borderBottomWidth": 1,
              "display": "flex",
              "flexDirection": "row",
              "height": 48,
              "paddingLeft": 20,
              "paddingRight": 20,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "display": "flex",
              },
              Object {
                "flex": 1,
                "height": 48,
              },
            ]
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                  Object {
                    "alignItems": "center",
                    "borderBottomColor": "#000",
                    "borderBottomWidth": 1,
                    "height": 48,
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <Text
                fontFamily="Unica77LL-Medium"
                fontSize="14px"
                lineHeight="24px"
                style={
                  Array [
                    Object {
                      "fontFamily": "Unica77LL-Medium",
                      "fontSize": 14,
                      "lineHeight": 24,
                    },
                    Object {},
                  ]
                }
              >
                About
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`layout works as expected with three tabs 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
      },
      Object {
        "flex": 1,
      },
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "overflow": "hidden",
        "position": "relative",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFF",
          "position": "absolute",
          "top": 0,
          "transform": Array [
            Object {
              "translateY": Object {
                "setValue": [Function],
              },
            },
          ],
          "width": 380,
        }
      }
    >
      <View
        onLayout={[Function]}
      >
        <ArtistHeader
          artist={
            Object {
              "birthday": null,
              "counts": Object {
                "artworks": 2,
                "partner_shows": 1,
              },
              "current_shows": null,
              "has_metadata": true,
              "id": null,
              "internalID": null,
              "name": null,
              "nationality": null,
              "past_large_shows": null,
              "past_small_shows": null,
              "upcoming_shows": null,
            }
          }
        />
        <View
          mb={1}
          style={
            Array [
              Object {
                "marginBottom": 10,
              },
            ]
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "#C2C2C2",
              "borderBottomWidth": 1,
              "display": "flex",
              "flexDirection": "row",
              "height": 48,
              "paddingLeft": 20,
              "paddingRight": 20,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "display": "flex",
              },
              Object {
                "flex": 1,
                "height": 48,
              },
            ]
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                  Object {
                    "alignItems": "center",
                    "borderBottomColor": "#C2C2C2",
                    "borderBottomWidth": 1,
                    "height": 48,
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <Text
                fontFamily="Unica77LL-Regular"
                fontSize="14px"
                lineHeight="24px"
                style={
                  Array [
                    Object {
                      "fontFamily": "Unica77LL-Regular",
                      "fontSize": 14,
                      "lineHeight": 24,
                    },
                    Object {},
                  ]
                }
              >
                About
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "display": "flex",
              },
              Object {
                "flex": 1,
                "height": 48,
              },
            ]
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                  Object {
                    "alignItems": "center",
                    "borderBottomColor": "#000",
                    "borderBottomWidth": 1,
                    "height": 48,
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <Text
                fontFamily="Unica77LL-Medium"
                fontSize="14px"
                lineHeight="24px"
                style={
                  Array [
                    Object {
                      "fontFamily": "Unica77LL-Medium",
                      "fontSize": 14,
                      "lineHeight": 24,
                    },
                    Object {},
                  ]
                }
              >
                Artworks
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "display": "flex",
              },
              Object {
                "flex": 1,
                "height": 48,
              },
            ]
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                  Object {
                    "alignItems": "center",
                    "borderBottomColor": "#C2C2C2",
                    "borderBottomWidth": 1,
                    "height": 48,
                    "justifyContent": "center",
                  },
                ]
              }
            >
              <Text
                fontFamily="Unica77LL-Regular"
                fontSize="14px"
                lineHeight="24px"
                style={
                  Array [
                    Object {
                      "fontFamily": "Unica77LL-Regular",
                      "fontSize": 14,
                      "lineHeight": 24,
                    },
                    Object {},
                  ]
                }
              >
                Shows
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
